src/solver/expressions/visitors/AstDOTStyleVisitor.cpp

```diff
@@ -0,0 +1,252 @@
+}
+
+void AstDOTStyleVisitor::visit(const Nodes::NegationNode* node, std::ostream& os)
+{
+    auto id = getNodeID(node);
+    emitNode(id, "-", NodeStyle::NegationStyle, os);
+    int childId = getNodeID(node->child());
+    os << "  " << id << " -> " << childId << ";\n";
+    dispatch(node->child(), os);
+}
+
+void AstDOTStyleVisitor::visit(const Nodes::PortFieldNode* node, std::ostream& os)
+{
+    auto id = getNodeID(node);
+    emitNode(id,
+             "PF(" + node->getPortName() + "," + node->getFieldName() + ")",
+             NodeStyle::PortFieldStyle,
+             os);
+}
+
+void AstDOTStyleVisitor::visit(const Nodes::ComponentVariableNode* node, std::ostream& os)
+{
+    auto id = getNodeID(node);
+    emitNode(id,
+             "CV(" + node->getComponentId() + "," + node->getComponentName() + ")",
+             NodeStyle::ComponentVariableStyle,
+             os);
+}
+
+void AstDOTStyleVisitor::visit(const Nodes::ComponentParameterNode* node, std::ostream& os)
+{
+    auto id = getNodeID(node);
+    emitNode(id,
+             "CP(" + node->getComponentId() + "," + node->getComponentName() + ")",
+             NodeStyle::ComponentParameterStyle,
+             os);
+}
+
+std::string AstDOTStyleVisitor::name() const
+{
+    return "AstGraphVisitor";
+}
+
+unsigned int AstDOTStyleVisitor::getNodeID(const Nodes::Node* node)
+{
+    const auto& node_name = node->name();
+    if (nodeIds_.find(node_name) == nodeIds_.end())
+    {
+        nodeIds_[node->name()][node] = ++nodeCount_;
+    }
+    else if (auto& id_map = nodeIds_[node_name]; id_map.find(node) == id_map.end())
+    {
+        id_map[node] = ++nodeCount_;
+    }
+
+    return nodeIds_[node->name()][node];
+}
+
+void AstDOTStyleVisitor::emitNode(unsigned int id,
+                                  const std::string& label,
+                                  const BoxStyle& box_style,
+                                  std::ostream& os)
+{
+    os << "  " << id << " [label=\"" << label << "\", shape=\"" << box_style.shape << "\", style=\""
+       << box_style.style << "\", color=\"" << box_style.color << "\"];\n";
+}
+
+// Process binary operation nodes like Add, Subtract, etc.
+void AstDOTStyleVisitor::processBinaryOperation(const Nodes::BinaryNode* node,
+                                                const std::string& label,
+                                                const BoxStyle& box_style,
+                                                std::ostream& os)
+{
+    auto id = getNodeID(node);
+    emitNode(id, label, box_style, os);
+
+    const Nodes::Node* left = node->left();
+    const Nodes::Node* right = node->right();
+
+    auto leftId = getNodeID(left);
+    auto rightId = getNodeID(right);
+
+    os << "  " << id << " -> " << leftId << ";\n";
+    os << "  " << id << " -> " << rightId << ";\n";
+
+    dispatch(left, os);
+    dispatch(right, os);
+}
+
+void AstDOTStyleVisitor::NewTreeGraph(std::ostream& os, const std::string& tree_name)
```

Comment:
`AstDOTStyleVisitor::NewTreeGraph` : this member function don't not use data member. 
Why make it a data member and not a free function ?

Reply:
see my [reply](https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/2399#discussion_r1766437874)

Reply:
If my reply addresses the issue, please mark this thread as resolved.